examples/auth.rs

```diff
@@ -0,0 +1,294 @@
+use std::fmt;
+use std::os::raw::c_int;
+use valkey_module::alloc::ValkeyAlloc;
+use valkey_module::{
+    valkey_module, Context, Status, ValkeyError, ValkeyString, AUTH_HANDLED, AUTH_NOT_HANDLED,
+};
+
+#[derive(Debug)]
+enum AuthResult {
+    Allow,
+    Deny,
+    Next,
+}
+
+#[derive(Debug)]
+struct AuthPrivData {
+    result: AuthResult,
+}
+
+// Implement Display for AuthResult
+impl fmt::Display for AuthResult {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            AuthResult::Allow => write!(f, "Allow"),
```

Comment:
can we do this instead?  

```
        match self {
            _ => write!(f, "{:?}", self),
        }

```

Reply:
You don't even need `match` in that case